### PR TITLE
[1.26] Fix socket timeout value when HTTPConnection is reused

### PR DESCRIPTION
**Pull request by nickwilliams-zaxiom**
This is a backport of PR #2792, for the purposes of applying the fix for issue #2645 to the 1.26.x release stream.

### CODE REVIEW

**Comment by pquentin** [comment]
I merged from master to include #2883 which will fix most CI failures. Sorry for the noise. Not sure about the app engine failure, but note that we dropped app engine in the `main` branch, which explains why it does not fail there.

**Comment by nickwilliams-zaxiom** [comment]
Thanks, @pquentin. That did, indeed, fix everything except `app_engine`. If we no longer support app_engine in `main`, I suggest we ignore that result here. But if I need to accommodate the `app_engine` failure, I think the easiest solution is to add `self.sock = None` to the constructor to mirror the CPython parent class constructor.

File: src/urllib3/connection.py [hunk 1]
```diff
@@ -229,6 +229,11 @@ def putheader(self, header, *values):
             )
 
     def request(self, method, url, body=None, headers=None):
+        # Update the inner socket's timeout value to send the request.
+        # This only triggers if the connection is re-used.
+        if self.sock is not None:
```
Comment (sethmlarson) [inline src/urllib3/connection.py:234]: To support AppEngine and other HTTPConnection implementations we can add `if getattr(self, "sock", None) is not None:`

File: changelog/2645.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Fixed an issue where an ``HTTPConnection`` instance would erroneously reuse the socket
```
Comment (sethmlarson) [inline changelog/2645.bugfix.rst:1]: We should delete this file since there's no `changelog/` directory on the 1.26.x branch.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thanks for opening this, we do need to support App Engine in the 1.26.x stream. See below for my suggestions:

File: src/urllib3/connection.py (hunk 1 above)
Comment (nickwilliams-zaxiom) [reply src/urllib3/connection.py]: Done

File: changelog/2645.bugfix.rst (hunk 2 above)
Comment (nickwilliams-zaxiom) [reply changelog/2645.bugfix.rst]: Done

**Comment by nickwilliams-zaxiom** [comment]
Hopefully everything passes this time. 🤞

**Comment by nickwilliams-zaxiom** [comment]
Would you look at that. All checks have passed. :-) Just need a 👍  from someone.

**Review by sethmlarson — APPROVED** [review]
Thanks much! :shipit: If you're interested in contributing more we've got lots of open issues, take a peek and see if you see one you're interested in :)

**Pull request merged**